android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/components/DateSelectionBox.kt

```diff
@@ -133,6 +136,31 @@ fun showDateRangePicker(
     }
 }
 
+fun setCustomReflection(
+  dpd: MaterialDatePicker.Builder<Pair<Long, Long>>
+): MaterialDatePicker.Builder<Pair<Long, Long>> {
+  try {
+    val datePickerDialogFields: Array<Field> = dpd::class.java.declaredFields
+    for (datePickerDialogField in datePickerDialogFields) {
+      if (datePickerDialogField.name == "mDatePicker") {
+        datePickerDialogField.isAccessible = true
+        val datePicker = datePickerDialogField[dpd] as DatePicker
+        val datePickerFields = datePickerDialogField.type.declaredFields
+        for (datePickerField in datePickerFields) {
+          if ("mDaySpinner" == datePickerField.name) {
+            datePickerField.isAccessible = true
+            val dayPicker = datePickerField[datePicker]
+            (dayPicker as View).visibility = View.GONE
+          }
+        }
+      }
+    }
+  } catch (ex: Exception) {
+    ex.printStackTrace()
```

Comment:
Use `Timber` instead to log errors. It would be great not to generalize the exception caught here. You can catch the specific exception instead.

Reply:
done with this .